src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -99,20 +112,56 @@ namespace {
 NetworkInfo sNetworks[kMaxNetworks];
 } // namespace
 
-// TODO: We cannot get length of operational dataset here, should be fixed after we have byte string type.
-// Note: The codegen for OCTET_STRING will be implemented later since no other clusters are using OCTET_STRING.
 EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId,
-                                                                           const uint8_t * operationalDataset, uint64_t breadcrumb,
+                                                                           const uint8_t * operationalDataset,
+                                                                           size_t operationalDatasetLen, uint64_t breadcrumb,
                                                                            uint32_t timeoutMs)
 {
-    // TODO: Implement AddThreadNetworkCommand after OCTET_STRING is supported.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_NETWORK_NOT_FOUND;
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            VerifyOrExit(operationalDatasetLen <= sizeof(ThreadNetworkInfo::mDataset),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            memcpy(sNetworks[i].mData.mThread.mDataset, operationalDataset, operationalDatasetLen);
+
+            VerifyOrExit(chip::CanCastTo<std::remove_cv_t<decltype(sNetworks[i].mData.mThread.mDatasetLen)>>(operationalDatasetLen),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
```

Comment:
I don't understand this check.  We're not castring to that long type; we're casting to `uint8_t`.  Maybe that type _is_ `uint8_t`, but then we should be casting to it.....

Point is: the type for `CanCastTo` should be the type we cast to.

Reply:
Found we don't need remove_cv_t here, thanks.